### DISCUSSION TITLE
Can an action get the path of the workflow event

### DISCUSSION THREAD

**Discussion by mudlabs** [community/community#26794] [Actions]
<p>Can an action get the path of the event that triggered the workflow to run?</p>
<p>Assume the workflow is as follows</p>
<pre><code class="lang-yaml">on:
  push:
    paths:
      - path/to/file

jobs:
  first:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - uses: user/log-event-path@v1.0.0
</code></pre>
<p>Would the <code>user/log-event-path</code> action be able to determine the event path by querying <code>@actions/core</code> or <code>@actions/github</code> etc…?</p>
<ul>
<li>The action runs using <code>node12</code>.</li>
</ul>

**Comment by mudlabs** [comment]
<p>I guess on push or pr events the action could find a file in the diff tree, and assume it’s the event path.</p>
<p>Then it’s the workflow developers responsibility to restrict the event to only trigger on one path.</p>
<p>Still it would be good if there was a <code>github.context.event.path</code> endpoint or something.</p>

**Comment by mudlabs** [comment]
<p>No that wouldn’t work (be reliable), because there could be multiple files in the one commit. The workflow would be triggered, but we still couldn’t be sure witch file corresponds to the event path.</p>
<p>Perhaps get the workflow file that triggered the action. Then parse it for the event path.</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/mudlabs">@mudlabs</a>,</p>
<p>In your custom JavaScript action, you can wrap the “<a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#get-a-commit" rel="noopener nofollow ugc"><strong>Get a commit</strong></a>” API.<br>
From the response context of this API, you can see all the changed files contained in this commit are listed in the “<strong>files[ ]</strong>” object (a JSON array).</p>
<p>If you want to list the files on a pull request, you can wrap the “<a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/pulls#list-pull-requests-files" rel="noopener nofollow ugc"><strong>List pull requests files</strong></a>” API in your action.</p>

**Comment by mudlabs** [comment]
<p>Yes but that does not help.</p>
<p>The issue is in determining which of the changed files is the file that triggered the workflow <em>(and the action</em>) to run.</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/mudlabs">@mudlabs</a>,</p>
<p>At first, you can list all the files contained in the specified commit.<br>
Then use the patterns you set on the path filters to compare with the path of each file in the list.<br>
The files that can match the patterns are the files that triggered the workflow run. The matched files may be one or more.</p>

**Comment by mudlabs** [comment]
<p>Yes, but I’m talking about a published action. If it was just for me I’d know what the path/s where already.</p>
<p>If github does not provide an <code>event_path</code> property to the action, I think the only option is as I mentioned above; Find and read the workflow file that called the action, parse it for any possible paths, and then filter against the commit file paths.</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/mudlabs">@mudlabs</a>,</p>
<blockquote>
<p>but I’m talking about a published action.</p>
</blockquote>
<p>For a published action, the authors had designed that what the action can do or not.</p>
<p>If you are not clear what things an action can do, you can view the README and the source codes of the action, or contact the authors to get more details information.</p>
<p>If you want to expand the features of an action, you can fork the repository of the action to your account and re-develop the action as your demands.</p>
<p>Again, if you want to view which files triggered the event, at first you need to list all the changed files contained in the commit, then compare with the patterns of the path filters.<br>
To get the list of the changed files, no matter in an action or directly in the workflow, you can run the API I mentioned above, or git commands (such as <a href="https://git-scm.com/docs/git-diff" rel="noopener nofollow ugc"><strong>git diff</strong></a>).</p>

**Comment by mudlabs** [comment]
<p>I am the actions author.</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/mudlabs">@mudlabs</a>,</p>
<blockquote>
<p>I am the actions author.</p>
</blockquote>
<p>You just need to improve the source code of the action to add the new features you require, then publish a new version of the action.</p>

**Comment by mudlabs** [comment]
<p>Ok thanks mate, I’ll take it from here.</p>

**Comment by gilbertbw** [comment]
<p>Hi <a class="mention" href="/mudlabs">@mudlabs</a>, did you figure out a way to get the paths from an action?</p>

**Comment by mudlabs** [comment, marked as answer]
<p><a class="mention" href="/gilbertbw">@gilbertbw</a> YES</p>
<pre><code class="lang-auto">const workflows = await octokit.request(
  "GET /repos/:owner/:repo/actions/workflows", 
  { owner, repo }
);
const workflow = workflows.data.workflows.find(
  workflow =&gt; workflow.name === process.env.GITHUB_WORKFLOW
);
</code></pre>

**Reply by jsoref** [reply]
No. This is *WRONG*.

Or rather, it's only right occasionally, and not for the workflow described above if the repository is private. this requires `actions: read`.
